In [5]:
import pandas as pd

In [8]:
data=pd.read_csv(r'C:\Users\Admin\StudentsPerformance.csv')
data.head()

gender race/ethnicity parental level of education         lunch  \
0  female        group C            some high school  free/reduced   
1  female        group B                 high school  free/reduced   
2  female        group B            some high school  free/reduced   
3  female        group B                some college      standard   
4  female        group C                some college  free/reduced   

  test preparation course  math score  reading score  writing score  
0                    none           0             17             10  
1                    none           8             24             23  
2                    none          18             32             28  
3                    none          11             38             32  
4                    none          22             39             33

# 1Participated Male and Female count

In [10]:
df=pd.DataFrame(data)
df["gender"].value_counts()

female    518
male      482
Name: gender, dtype: int64

The number of females participated = 518
The number of males participated = 482

# 2Student's Parent Level of Education

In [12]:
df["parental level of education"].value_counts()

some college          226
associate's degree    222
high school           196
some high school      179
bachelor's degree     118
master's degree        59
Name: parental level of education, dtype: int64

Most of the student's parents have some college level of education.Only 59 parents have Master's Degree.

# 3Scores the Most on Average

*Gender

In [20]:
A=data.groupby(["gender"]).mean()
B=A['math score'].sort_values(ascending=False)
print('The gender has the highest math score is ')
B

The gender has the highest math score is 


gender
male      68.821577
female    63.376448
Name: math score, dtype: float64

In Maths, Males scores the most average.

In [21]:
C=A['reading score'].sort_values(ascending=False)
print('The gender has the highest reading score is ')
C

The gender has the highest reading score is 


gender
female    72.590734
male      65.545643
Name: reading score, dtype: float64

In Reading,Female scores the most average.

In [22]:
D=A['writing score'].sort_values(ascending=False)
print('The gender has the highest writing score is  ')
D

The gender has the highest writing score is  


gender
female    72.467181
male      63.446058
Name: writing score, dtype: float64

In Writing,Females scores the most average.

In [23]:
sum=B+C+D
sum

gender
female    208.434363
male      197.813278
dtype: float64

Therefore in gender wise females scores the most average.

*Test Prepration Course

In [24]:
E=data.groupby(["test preparation course"]).mean()
F=E['math score']
F

test preparation course
completed    69.96648
none         63.78972
Name: math score, dtype: float64

In [25]:
G=E['reading score']
G

test preparation course
completed    74.175978
none         66.417445
Name: reading score, dtype: float64

In [26]:
H=E['writing score']
H

test preparation course
completed    74.684358
none         64.457944
Name: writing score, dtype: float64

In [27]:
sum=F+G+H
sum

test preparation course
completed    218.826816
none         194.665109
dtype: float64

Most students completed the test prepration course.

# 4Scoring Variations

*Gender

In [28]:
I=data.groupby(["gender"]).std()
J=I['math score'].sort_values()
print('math score is ')
J

math score is 


gender
male      14.556411
female    16.029928
Name: math score, dtype: float64

In [29]:
K=I['reading score'].sort_values()
print('reading score is ')
K

reading score is 


gender
male      14.149594
female    14.411018
Name: reading score, dtype: float64

In [30]:
L=I['writing score'].sort_values()
print('writing score is ')
L

writing score is 


gender
male      14.227225
female    14.844842
Name: writing score, dtype: float64

In [31]:
sum=J+K+L
sum

gender
male      42.933230
female    45.285788
dtype: float64

*Test Prepration Course

In [32]:
M=data.groupby(["test preparation course"]).std()
N=M['math score']
N

test preparation course
completed    14.521847
none         15.705689
Name: math score, dtype: float64

In [33]:
O=M['reading score']
O

test preparation course
completed    13.537572
none         14.608896
Name: reading score, dtype: float64

In [34]:
P=M['writing score']
P

test preparation course
completed    13.236412
none         15.041667
Name: writing score, dtype: float64

In [35]:
sum=N+O+P
sum

test preparation course
completed    41.295831
none         45.356252
dtype: float64

# 5Bonus Points

In [37]:
import numpy as np

In [38]:
Third_quartile=np.quantile(data['math score'],0.75)
Third_quartile

77.0

In [39]:
math_score=data.sort_values(by=['math score'],ascending=False)
n=25
math_score.head(int(len(math_score)*(n/100)))

gender race/ethnicity parental level of education         lunch  \
999    male        group E           bachelor's degree      standard   
996    male        group A                some college      standard   
515  female        group E                some college      standard   
517  female        group E          associate's degree      standard   
516  female        group E           bachelor's degree      standard   
..      ...            ...                         ...           ...   
856    male        group E            some high school      standard   
855    male        group E          associate's degree  free/reduced   
854    male        group D            some high school      standard   
853    male        group D          associate's degree  free/reduced   
852    male        group D                some college  free/reduced   

    test preparation course  math score  reading score  writing score  
999               completed         100            100            100  
996               completed         100             96             86  
515                    none         100             92             97  
517                    none         100            100            100  
516                    none         100            100            100  
..                      ...         ...            ...            ...  
856               completed          77             76             77  
855               completed          77             69             68  
854               completed          77             68             69  
853                    none          77             78             73  
852                    none          77             62             64  

[250 rows x 8 columns]